filters/include/bloom_filter.hpp

```diff
@@ -0,0 +1,757 @@
+class bloom_filter_alloc {
+  using A = Allocator;
+
+public:
+
+  /**
+   * This method deserializes a Bloom filter from a given array of bytes.
+   * @param bytes pointer to the array of bytes
+   * @param size the size of the array
+   * @param allocator instance of an Allocator
+   * @return an instance of a Bloom filter
+   */
+  static bloom_filter_alloc deserialize(const void* bytes, size_t length_bytes, const Allocator& allocator = Allocator());
+
+  /**
+   * This method deserializes a Bloom filter from a given stream.
+   * @param is input stream
+   * @param allocator instance of an Allocator
+   * @return an instance of a Bloom filter
+   */
+  static bloom_filter_alloc deserialize(std::istream& is, const A& allocator = Allocator());
+
+  /**
+   * @brief Wraps the provided memory as a read-only Bloom filter. Reads the data in-place and does
+   * not take ownership of the underlying memory. Does not allow modifying the filter.
+   *
+   * @param data The memory to wrap
+   * @param length_bytes The length of the memory in bytes
+   * @param allocator instance of an Allocator
+   * @return a const (read-only) Bloom filter wrapping the provided memory
+   */
+  static const bloom_filter_alloc wrap(const void* data, size_t length_bytes, const Allocator& allocator = Allocator());
+
+  /**
+   * @brief Wraps the provided memory as a writable Bloom filter. Reads the data in-place and does
+   * not take ownership of the underlying memory. Allows modifying the filter.
+   *
+   * @param data the memory to wrap
+   * @param length_bytes the length of the memory in bytes
+   * @param allocator instance of an Allocator
+   * @return a Bloom filter wrapping the provided memory
+   */
+  static bloom_filter_alloc writable_wrap(void* data, size_t length_bytes, const Allocator& allocator = Allocator());
+
+  /**
+   * Copy constructor
+   * @param other filter to be copied
+   */
+  bloom_filter_alloc(const bloom_filter_alloc&);
+
+  /** Move constructor
+   * @param other filter to be moved
+   */
+  bloom_filter_alloc(bloom_filter_alloc&&) noexcept;
+
+  /**
+   * Copy assignment
+   * @param other filter to be copied
+   * @return reference to this filter
+   */
+  bloom_filter_alloc& operator=(const bloom_filter_alloc& other);
+
+  /**
+   * Move assignment
+   * @param other filter to be moved
+   * @return reference to this filter
+   */
+  bloom_filter_alloc& operator=(bloom_filter_alloc&& other);
+
+  /**
+   * @brief Destroy the bloom filter alloc object
+   */
+  ~bloom_filter_alloc();
+
+  // This is a convenience alias for users
+  // The type returned by the following serialize method
+  using vector_bytes = std::vector<uint8_t, typename std::allocator_traits<A>::template rebind_alloc<uint8_t>>;
+
+  /**
+   * This method serializes the filter as a vector of bytes.
+   * An optional header can be reserved in front of the filter.
+   * It is a blank space of a given size.
+   * This header is used in Datasketches PostgreSQL extension.
```

Comment:
not sure we want to mention PostgreSQL here

Reply:
We did on all other sketches?

Reply:
I think we did. but I am not sure it was good, especially for a sketch that is not supported in PSQL
perhaps we could soften the statement "is used". something like "such headers might be needed in some systems like PSQL"

Reply:
How is "Some integrations such as PostgreSQL may need this header space."

Reply:
looks fine